source/random/engine/pcg.d

```diff
@@ -0,0 +1,449 @@
+/**
+ * Permuted Congruential Generator (PCG)
+ *
+ * Implemeted as per the C++ version of PCG. see $(HTTP www.pcg-random.org)
+ */
+
+module random.engine.pcg;
+
+import random.engine;
+import core.bitop;
+
+@safe:
+nothrow:
+@nogc:
+
+private template default_multiplier(T)
+{
+    static if (is(T == ubyte))
+        enum ubyte default_multiplier = 141u;
+    else static if (is(T == ushort))
+        enum ushort default_multiplier = 12829u;
+    else static if (is(T == uint))
+        enum uint default_multiplier = 747796405u;
+    else static if (is(T == ulong))
+        enum ulong default_multiplier = 6364136223846793005u;
+    //TODO: ucent PCG_128BIT_CONSTANT(2549297995355413924ULL,
+    //                                4865540595714422341ULL)
+    else
+        static assert(0);
+}
+
+private template default_increment(T)
+{
+    static if (is(T == ubyte))
+        enum ubyte default_increment = 77u;
+    else static if (is(T == ushort))
+        enum ushort default_increment = 47989u;
+    else static if (is(T == uint))
+        enum uint default_increment = 2891336453u;
+    else static if (is(T == ulong))
+        enum ulong default_increment = 1442695040888963407u;
+    //TODO: ucent PCG_128BIT_CONSTANT(6364136223846793005ULL,
```

Comment:
Do those actually work? I was going to do an ulong[2] pair, there's an implementation lying around somewhere.

Also don't hesitate to edit the PR.